src/CSEjector.sol

```diff
@@ -60,6 +60,41 @@ contract CSEjector is
         _pauseFor(duration);
     }
 
+    function voluntaryEject(uint256 nodeOperatorId, uint256 keyIndex) public {
+        NodeOperatorManagementProperties memory no = MODULE
+            .getNodeOperatorManagementProperties(nodeOperatorId);
+        if (no.managerAddress == address(0)) {
+            revert NodeOperatorDoesNotExist();
+        }
+        if (no.managerAddress != msg.sender && no.rewardAddress != msg.sender) {
+            revert SenderIsNotEligible();
+        }
```

Comment:
I think we should account for the extended manager permissions here as we did in bond curve claim